packages/astro/test/units/routing/manifest.test.js

```diff
@@ -29,7 +37,173 @@ describe('routing - createRouteManifest', () => {
 		expect(pattern.test('/')).to.equal(false);
 	});
 
-	it('redirects are sorted alongside the filesystem routes', async () => {
+	it('endpoint routes are sorted before page routes', async () => {
+		const fs = createFs(
+			{
+				'/src/pages/contact-me.astro': `<h1>test</h1>`,
+				'/src/pages/sendContact.ts': `<h1>test</h1>`,
+			},
+			root,
+		);
+		const settings = await createBasicSettings({
+			root: fileURLToPath(root),
+			base: '/search',
+			trailingSlash: 'never',
+		});
+
+		settings.injectedRoutes = [
+			{
+				pattern: '/about',
+				entrypoint: '@lib/legacy/static.astro',
+			},
+			{
+				pattern: '/api',
+				entrypoint: '@lib/legacy/static.ts',
+			},
+		];
+
+		const manifest = createRouteManifest({
+			cwd: fileURLToPath(root),
+			settings,
+			fsMod: fs,
+		});
+
+		expect(getManifestRoutes(manifest)).to.deep.equal([
+			{
+				route: '/api',
+				type: 'endpoint',
+			},
+			{
+				route: '/sendcontact',
+				type: 'endpoint',
+			},
+			{
+				route: '/about',
+				type: 'page',
+			},
+			{
+				route: '/contact-me',
+				type: 'page',
+			},
+		]);
+	});
+
+	it('injected routes are sorted in legacy mode above filesystem routes', async () => {
+		const fs = createFs(
+			{
+				'/src/pages/index.astro': `<h1>test</h1>`,
+				'/src/pages/blog/[...slug].astro': `<h1>test</h1>`,
+			},
+			root,
+		);
+		const settings = await createBasicSettings({
+			root: fileURLToPath(root),
+			output: 'server',
+			base: '/search',
+			trailingSlash: 'never',
+		});
+
+		settings.injectedRoutes = [
+			{
+				pattern: '/contributing',
+				entrypoint: '@lib/legacy/static.astro',
+			},
+			{
+				pattern: '/[...slug]',
+				entrypoint: '@lib/legacy/dynamic.astro',
+				priority: 'legacy',
+			},
+		];
+
+		const manifest = createRouteManifest({
+			cwd: fileURLToPath(root),
+			settings,
+			fsMod: fs,
+		});
+
+		expect(getManifestRoutes(manifest)).to.deep.equal([
+			{
+				route: '/[...slug]',
```

Comment:
Is this route at the first place because it's a "legacy" route?

Reply:
Yes, now it is the second because both injected routes are legacy